### PR TITLE
Add an endpoint for creating new users

### LINKED ISSUE #1088: Add user endpoint (and the corr. UI) needed
Needs for this endpoint - probably for "superuser's" usage only - have been expressed.  If there's no-one working with this at the moment, I could volunteer. Later on, it could be expanded to a full CRUD for users.

I've already verified how user addition is done on Keycloak REST API, and seems like there are readymade functions [here](https://github.com/eclipse-apoapsis/ort-server/blob/f34f99d591116430a57a0f63f5c38249772c9046/clients/keycloak/src/main/kotlin/DefaultKeycloakClient.kt#L305-L378).

### PR DESCRIPTION
**Pull request by Etsija**
Please see the commits for details.
A subsequent PR will implement a minimal admin UI for creating new users.

Resolves #1088.

### CODE REVIEW

File: api/v1/model/src/commonMain/kotlin/User.kt [hunk 1]
```diff
@@ -27,7 +27,14 @@ import kotlinx.serialization.Serializable
...
     val id: String,
-    val username: String
+    val username: String,
+)
+
+@Serializable
+data class CreateUser(
+    val username: String,
+    val password: String? = null,
+    val temporary: Boolean = true
```
Comment (mmurto) [inline api/v1/model/src/commonMain/kotlin/User.kt:37]: This and OpenAPI should document what temporary does.

**Review by mmurto — CHANGES_REQUESTED** [review]

File: clients/keycloak/src/main/kotlin/KeycloakClient.kt [hunk 2]
```diff
@@ -127,15 +127,23 @@ interface KeycloakClient {
...
+     * [password] and [temporary].
      */
     suspend fun createUser(
         username: UserName,
         firstName: String? = null,
         lastName: String? = null,
-        email: String? = null
+        email: String? = null,
+        password: String? = null,
+        temporary: Boolean = false
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply clients/keycloak/src/main/kotlin/KeycloakClient.kt:139]: While it'd be possible to model that with a dedicated `Password` class or some `require` checks, personally I'd be fine with just documenting that `temporary` is ignores with `password` is `null`.

File: api/v1/model/src/commonMain/kotlin/User.kt (hunk 1 above)
Comment (Etsija) [reply api/v1/model/src/commonMain/kotlin/User.kt:39]: @murto I couldn't find any examples from `ktorswaggerui` where the descriptions could be given at the field level, only at body level.

File: clients/keycloak/src/main/kotlin/KeycloakClient.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Etsija) [reply clients/keycloak/src/main/kotlin/KeycloakClient.kt:139]: Done.

**Comment by Etsija** [comment]
@mnonnenmacher @sschuberth @mmurto could I kindly ask for a review decision on this PR? I'd like to know whether it still needs rework, or whether I can move on to something else (like, add a very minimal UI for it or start working on UI implementation for the new endpoints).

File: api/v1/model/src/commonMain/kotlin/User.kt (hunk 1 above)
Comment (mnonnenmacher) [reply api/v1/model/src/commonMain/kotlin/User.kt:39]: I just noticed that my answer above was stuck in pending state since yesterday...

Comment (Etsija) [reply api/v1/model/src/commonMain/kotlin/User.kt:39]: @mnonnenmacher I'm sorry but I lack the knowledge of how to use that. Is it a library that needs to be imported from somewhere?  The example you linked is lacking on some crucial info (at least to me), and as such, IntelliJ doesn't recognize it, so obviously the `User.kt` file is lacking an import.

**Comment by mnonnenmacher** [comment]
> @mnonnenmacher @sschuberth @mmurto could I kindly ask for a review decision on this PR? I'd like to know whether it still needs rework, or whether I can move on to something else (like, add a very minimal UI for it or start working on UI implementation for the new endpoints).

After looking at the PR again I just wonder, why did you not add the new routes to the existing admin route?

**Comment by Etsija** [comment]
> > @mnonnenmacher @sschuberth @mmurto could I kindly ask for a review decision on this PR? I'd like to know whether it still needs rework, or whether I can move on to something else (like, add a very minimal UI for it or start working on UI implementation for the new endpoints).
> 
> After looking at the PR again I just wonder, why did you not add the new routes to the existing admin route?

Can't think of a reason why not. We had an internal team review and wanted to get some comments from you as to that.  Another thing is, I'm not sure `UserService` belongs under `hierarchy`, either.

File: api/v1/model/src/commonMain/kotlin/User.kt (hunk 1 above)
Comment (mnonnenmacher) [reply api/v1/model/src/commonMain/kotlin/User.kt:39]: To me that's not mandatory, we do not use that anywhere else.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/apiDocs/AdminDocs.kt [hunk 3]
```diff
@@ -41,3 +43,29 @@ val runPermissionsSync: OpenApiRoute.() -> Unit = {
...
+            }
+        }
+    }
+
+    response {
+        HttpStatusCode.NoContent to {
+            description = "Successfully created the user."
+        }
+
+        HttpStatusCode.Conflict to {
```
[the remark this answers is not in the document]
Comment (Etsija) [reply core/src/main/kotlin/apiDocs/AdminDocs.kt]: Actually, as was mentioned in the PR where the organization group handling was introduced, Keycloak doesn't distinguish between technical and logical errors, so a general "500" is propagated through the layers in error cases. This is now fixed and documented.

**Pull request merged**